packages/caravan-health/src/feescore.ts

```diff
@@ -0,0 +1,145 @@
+import { BlockchainClient } from "@caravan/clients";
+import { utxoSetLengthScore } from "./privacy";
+import { Transaction, AddressUtxos } from "./types";
+
+/*
+Utility function that helps to obtain the fee rate of the transaction
+
+Expected Range : [0, 0.75]
+-> Very Poor : [0, 0.15]
+-> Poor : (0.15, 0.3]
+-> Moderate : (0.3, 0.45]
+-> Good : (0.45, 0.6]
+-> Very Good : (0.6, 0.75]
+*/
+function getFeeRateForTransaction(transaction: Transaction): number {
+  let fees: number = transaction.fee;
+  let weight: number = transaction.weight;
+  return fees / weight;
+}
+
+/* 
+Utility function that helps to obtain the percentile of the fees paid by user in tx block
+
+Expected Range : [0, 0.75]
+-> 0% tile : 1
+-> 10% tile : 0.9
+-> 25% tile : 0.75
+-> 50% tile : 0.5
+-> 75% tile : 0.25
+-> 90% tile : 0.1
+-> 100% tile : 0.05
+*/
+async function getFeeRatePercentileScore(
+  timestamp: number,
+  feeRate: number,
+  client: BlockchainClient,
+) {
+  let percentile: number = await client.getFeeRatePercentileForTransaction(
+    timestamp,
+    feeRate,
+  );
+  switch (percentile) {
+    case 0:
+      return 1;
+    case 10:
+      return 0.9;
+    case 25:
+      return 0.75;
+    case 50:
+      return 0.5;
+    case 75:
+      return 0.25;
+    case 90:
+      return 0.1;
+    case 100:
+      return 0;
+    default:
+      throw new Error("Invalid percentile");
+  }
+}
+
+/*
+R.F.S can be associated with all the transactions and we can give a measure 
+if any transaction was done at expensive fees or nominal fees.
+
+This can be done by calculating the percentile of the fees paid by the user
+in the block of the transaction.
+
+Expected Range : [0, 1]
+-> Very Poor : [0, 0.2]
+-> Poor : (0.2, 0.4]
+-> Moderate : (0.4, 0.6]
+-> Good : (0.6, 0.8]
+-> Very Good : (0.8, 1]
+*/
+export async function relativeFeesScore(
+  transactions: Transaction[],
+  client: BlockchainClient,
+): Promise<number> {
+  let sumRFS: number = 0;
+  let numberOfSendTx: number = 0;
+  for (const tx of transactions) {
+    if (tx.isSend === true) {
+      numberOfSendTx++;
+      let feeRate: number = getFeeRateForTransaction(tx);
+      let RFS: number = await getFeeRatePercentileScore(
```

Comment:
this definitely makes me think we should do some minimal cacheing. The way things are written now, you will be requesting the full payload for every transaction, when really you should only need to make the request once and then get the percentile score. Imagine what this looks like for any wallet with a decent tx history making this call 40-50 times. 

One way would be to just have the client do a call to get the block fee rates, something like `client.getBlockFeeRates` then `relativeFeesScore` could just take the result as an arg or you can have another intermediate function to do the calculation for you. Another option would be to cache the result on the client class itself and if it has a value for block fee rates then use that, if not make the request and store it on the instance for later.